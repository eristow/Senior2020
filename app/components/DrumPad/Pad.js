import styled from 'styled-components';

const Pad = styled.button`
  display: inline-block;
  box-sizing: border-box;
  height: 100px;
  width: 100px;
  padding: 0.25em 1em;
  margin: 0em 0.1em;
  text-decoration: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: bold;
  font-size: 16px;
  border: 2px solid #cc4141;
  color: #cc4141;
  background: #fff;

  &:active {
    background: #cc4141;
    color: #fff;
  }
`;

export default Pad;
