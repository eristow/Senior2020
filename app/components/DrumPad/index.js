/**
 *
 * DrumPad
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';

import Pad from './Pad';

function DrumPad(props) {
  const audio = new Audio(props.src);

  const handleClick = () => {
    audio.play();
  };

  return (
    <div>
      <Pad onClick={handleClick}>{props.name}</Pad>
    </div>
  );
}

DrumPad.propTypes = {
  name: PropTypes.string,
  src: PropTypes.string,
};

export default DrumPad;
