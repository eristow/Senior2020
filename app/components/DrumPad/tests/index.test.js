/**
 *
 * Tests for DrumPad
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import DrumPad from '../index';

describe('<DrumPad />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<DrumPad />);
    expect(spy).not.toHaveBeenCalled();
  });

  // TODO: add tests
  it.skip('Expect to have additional unit tests specified', () => {
    expect(true).toEqual(false);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<DrumPad />);
    expect(firstChild).toMatchSnapshot();
  });
});
