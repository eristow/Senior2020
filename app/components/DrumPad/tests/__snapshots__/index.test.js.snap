// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrumPad /> Should render and match the snapshot 1`] = `
<div>
  <button
    class="Pad-sc-1ad3ed0-0 cMOtpk"
  />
</div>
`;
